controllers/update_status.go

```diff
@@ -539,6 +546,13 @@ func validateInstances(r *FoundationDBClusterReconciler, context ctx.Context, cl
 		}
 	}
 
+	for _, processGroupStatus := range processGroups {
```

Comment:
This case happens when the Pod of a process group is removed but not all other resources like the PVC or service right? Could we add a short comment above that loop?